### PR TITLE
Added function to get the total size of directory if the user choose the option file_type: directory

### PR DESCRIPTION
**Pull request by ericcontainer**
##### SUMMARY

Get information of total size of directory. If the user choose file_type the new feature will return a total size, instead to return the minimum block size disk.
   When file_type is "file" the size is returned, but when the option is "directory" return 4096

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME
statinfo
```

```

### CODE REVIEW

**Comment by prune998** [comment]
👍 for this feature, I need exactly this right now :)

**Comment by v1sion** [comment]
need exactly this 👍

**Comment by desimaniac** [comment]
Status ?

File: lib/ansible/modules/files/find.py [hunk 1]
```diff
@@ -153,6 +153,11 @@
     returned: success
     type: string
     sample: 34
+total_size_dir:
+    description: total size of filesystem directory looked at
+    returned: success
+    type: int
+    sample: 50098
```
Comment (bcoca) [inline lib/ansible/modules/files/find.py:160]: add 'version_added'

**Review by bcoca — CHANGES_REQUESTED** [review]
I'm unsure this should be the default since it can greatly impact performance, I would add a option to return this data

**Comment by roxor45** [comment]
I find a bug when i try to get size directory of / : 

`        find:
          paths: '/'
          file_type: directory
        register: directory_size`

output: 
`The full traceback is:
Traceback (most recent call last):
  File "/root/.ansible/tmp/ansible-tmp-1575012377.33-60331210977224/AnsiballZ_find.py", line 114, in <module>
    _ansiballz_main()
  File "/root/.ansible/tmp/ansible-tmp-1575012377.33-60331210977224/AnsiballZ_find.py", line 106, in _ansiballz_main
    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
  File "/root/.ansible/tmp/ansible-tmp-1575012377.33-60331210977224/AnsiballZ_find.py", line 49, in invoke_module
    imp.load_module('__main__', mod, module, MOD_DESC)
  File "/tmp/ansible_find_payload_gsveX1/__main__.py", line 386, in <module>
  File "/tmp/ansible_find_payload_gsveX1/__main__.py", line 356, in main
  File "/tmp/ansible_find_payload_gsveX1/__main__.py", line 276, in statinfo
  File "/tmp/ansible_find_payload_gsveX1/__main__.py", line 241, in size_dir
  File "/tmp/ansible_find_payload_gsveX1/__main__.py", line 241, in <genexpr>
  File "/usr/lib64/python2.7/genericpath.py", line 49, in getsize
    return os.stat(filename).st_size
OSError: [Errno 2] No such file or directory: '/proc/2/exe'

fatal: [localhost]: FAILED! => {
    "changed": false,
    "module_stderr": "Traceback (most recent call last):\n  File \"/root/.ansible/tmp/ansible-tmp-1575012377.33-60331210977224/AnsiballZ_find.py\", line 114, in <module>\n    _ansiballz_main()\n  File \"/root/.ansible/tmp/ansible-tmp-1575012377.33-60331210977224/AnsiballZ_find.py\", line 106, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/root/.ansible/tmp/ansible-tmp-1575012377.33-60331210977224/AnsiballZ_find.py\", line 49, in invoke_module\n    imp.load_module('__main__', mod, module, MOD_DESC)\n  File \"/tmp/ansible_find_payload_gsveX1/__main__.py\", line 386, in <module>\n  File \"/tmp/ansible_find_payload_gsveX1/__main__.py\", line 356, in main\n  File \"/tmp/ansible_find_payload_gsveX1/__main__.py\", line 276, in statinfo\n  File \"/tmp/ansible_find_payload_gsveX1/__main__.py\", line 241, in size_dir\n  File \"/tmp/ansible_find_payload_gsveX1/__main__.py\", line 241, in <genexpr>\n  File \"/usr/lib64/python2.7/genericpath.py\", line 49, in getsize\n    return os.stat(filename).st_size\nOSError: [Errno 2] No such file or directory: '/proc/2/exe'\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    "rc": 1
}`

**Comment by fishhead108** [comment]
any news?

**Comment by jborean93** [comment]
Question About This One
=======================

Hi!

Thanks very much for your submission to Ansible.  It sincerely means a lot to us that you've taken time to contribute.

Unfortunately, we're not sure if we want this feature in the program, and I don't want this to seem confrontational.  Our reasons for this are:

   * Calculating the size of a directory is fraught with danger around performance and even just logic when it comes to special file types like symlink/hard links.

However, we're absolutely always up for discussion.  Since this is a really busy project, we don't always see comments on closed tickets, but want to encourage
open dialog.  You can stop by the development list, and we'd be glad to talk about it - and we might even be persuaded otherwise!

   * https://groups.google.com/forum/#!forum/ansible-devel

In the future, sometimes starting a discussion on the development list prior to implementing a feature can make getting things included a little easier, but it's not always necessary.

Thank you once again for this and your interest in Ansible!

**Pull request closed without merging**